Web Inspector: Debugger: blackboxing should default to case sensitive

dcrousso opened the pull request:
#### 6e462daaae46b9dca6817e61b308243175a8add6
<pre>
Web Inspector: Debugger: blackboxing should default to case sensitive
<a href="https://bugs.webkit.org/show_bug.cgi?id=290594">https://bugs.webkit.org/show_bug.cgi?id=290594</a>

Reviewed by BJ Burg.

This matches breakpoints and local overrides and searching and etc., as well as developer expectations (i.e. case insensitivity is an opt-in).

* Source/JavaScriptCore/inspector/protocol/Debugger.json:
* Source/JavaScriptCore/inspector/agents/InspectorDebuggerAgent.h:
* Source/JavaScriptCore/inspector/agents/InspectorDebuggerAgent.cpp:
(Inspector::InspectorDebuggerAgent::setShouldBlackboxURL):

* LayoutTests/inspector/debugger/resources/blackbox.js:
(ProtocolTest.Blackbox.setBlackbox):
* LayoutTests/inspector/debugger/setBlackboxBreakpointEvaluations.html:
* LayoutTests/inspector/debugger/setBlackboxBreakpointEvaluations-expected.txt:
* LayoutTests/inspector/debugger/setShouldBlackboxURL-exact-case-insensitive.html:
* LayoutTests/inspector/debugger/setShouldBlackboxURL-exact-case-sensitive.html:
* LayoutTests/inspector/debugger/setShouldBlackboxURL-exact-case-sensitive-expected.txt:
* LayoutTests/inspector/debugger/setShouldBlackboxURL-regex-case-insensitive.html:
* LayoutTests/inspector/debugger/setShouldBlackboxURL-regex-case-sensitive.html:
* LayoutTests/inspector/debugger/setShouldBlackboxURL-regex-case-sensitive-expected.txt:

Canonical link: <a href="https://commits.webkit.org/292997@main">https://commits.webkit.org/292997@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/72aa8ab56a4b6406821e318df5cef606fa2d71f5

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/97627 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/17252 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/7468 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/102733 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/48156 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/17546 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/25705 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/74374 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/31553 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/100630 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/13284 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/88275 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/54723 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/13056 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/6171 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/47598 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/90303 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/83078 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/6247 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/104734 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/96249 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/24707 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/18017 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/83421 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/25079 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/84413 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/82850 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20859 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/27402 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/5094 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/18302 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/24668 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/29837 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/119875 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/24490 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/33650 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/27804 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/26064 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

burg commented on LayoutTests/inspector/debugger/setBlackboxBreakpointEvaluations-expected.txt:
```diff
@@ -114,7 +114,7 @@ PASS: Should not pause in 'BlackboxBreakpointEvaluations_BreakpointAction_Pause_
-Blackboxing 'blackboxbreakpointevaluations_breakpointaction_autocontinue_inner.js'...
+Blackboxing 'BlackboxBreakpointEvaluations_BreakpointAction_AutoContinue_Inner.js'...
 Setting breakpoint in 'BlackboxBreakpointEvaluations_BreakpointAction_AutoContinue_Inner.js'...
 Blackboxing preakpoing evaluations...
```
WTF is `preoakpoing`? Can you fix this typo (probably `breakpoint`)?

burg reviewed (approved):
Thanks!

dcrousso committed:
Web Inspector: Debugger: blackboxing should default to case sensitive
https://bugs.webkit.org/show_bug.cgi?id=290594

Reviewed by BJ Burg.

This matches breakpoints and local overrides and searching and etc., as well as developer expectations (i.e. case insensitivity is an opt-in).

* Source/JavaScriptCore/inspector/protocol/Debugger.json:
* Source/JavaScriptCore/inspector/agents/InspectorDebuggerAgent.h:
* Source/JavaScriptCore/inspector/agents/InspectorDebuggerAgent.cpp:
(Inspector::InspectorDebuggerAgent::setShouldBlackboxURL):

* LayoutTests/inspector/debugger/resources/blackbox.js:
(ProtocolTest.Blackbox.setBlackbox):
* LayoutTests/inspector/debugger/setBlackboxBreakpointEvaluations.html:
* LayoutTests/inspector/debugger/setBlackboxBreakpointEvaluations-expected.txt:
* LayoutTests/inspector/debugger/setShouldBlackboxURL-exact-case-insensitive.html:
* LayoutTests/inspector/debugger/setShouldBlackboxURL-exact-case-sensitive.html:
* LayoutTests/inspector/debugger/setShouldBlackboxURL-exact-case-sensitive-expected.txt:
* LayoutTests/inspector/debugger/setShouldBlackboxURL-regex-case-insensitive.html:
* LayoutTests/inspector/debugger/setShouldBlackboxURL-regex-case-sensitive.html:
* LayoutTests/inspector/debugger/setShouldBlackboxURL-regex-case-sensitive-expected.txt:

Canonical link: https://commits.webkit.org/292997@main
